src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,72 @@
-import React from 'react';
+import React, { ChangeEvent, memo, useContext, useState } from 'react';
+import cn from 'classnames';
+import { PostsContext } from '../store/PostsContext';
+
+import { CommentsContext } from '../store/CommentsContext';
+
+export const NewCommentForm = memo(function NewCommentFormComponent() {
+  const { selectedPost } = useContext(PostsContext);
+  const { text, setText, addComment, addLoading } = useContext(CommentsContext);
+
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+
+  const [nameError, setNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [bodyError, setbodyError] = useState(false);
+
+  const handleNameChange = (
+    event: ChangeEvent<HTMLInputElement> | ChangeEvent<HTMLTextAreaElement>,
+    setValue: (value: string) => void,
+    setValueError: (error: boolean) => void,
+  ) => {
+    setValue(event.target.value);
+    setValueError(false);
+  };
+
+  const clear = () => {
+    setName('');
+    setNameError(false);
+    setEmail('');
+    setEmailError(false);
+    setText('');
+    setbodyError(false);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    const nameNormalize = name.trim();
```

Comment:
```suggestion
    const nameNormalized = name.trim();
```
it's not a function but already normalized value